### DISCUSSION TITLE
Export and Import EEPROM configuration values

### DISCUSSION THREAD

**Discussion by davkvi** [mriscoc/Ender3V2S1#38] [Ideas]
Hi,

One feature that would be a "nice to have" is to export the parameters set to the sd card.
e-steps, home pos and so on.

So we could import it to the firmware when we upgrade :)

**Comment by mriscoc** [comment]
Hi, I received this request from day 1, also the official Marlin and other fork. I know that it could be easy for the user to simply export a configuration file and then import after a flash, but from a programming point of view there is a penalty in the space that this feature will consume in the now reduced (RCT6) flash space, is not only about of export the values, also when importing they must pass a filter and verification routines for ensure that all parameters are within the workable range, that could double what we already have in almost all g-code commands.

In professional use, it is better if you create a special configuration g-code file with all the machine specifications that you need, then, after flashing a new firmware, you need only to load that file to restore your preferences. The Professional firmware implements the standard Marlin g-codes and also special custom g-codes to allow you to create a complete configuration file. I invite you to try to use this approach. Your configuration g-code file could be:
 ```g-code
M502              ; load the firmware default values
; put here what you need -----------------------------------------
M201 X500.00 Y500.00 Z100.00 E5000.00 ;Setup machine max acceleration
M203 X500.00 Y500.00 Z10.00 E50.00 ;Setup machine max feedrate
M204 P500.00 R1000.00 T500.00 ;Setup Print/Retract/Travel acceleration
M205 X8.00 Y8.00 Z0.40 E5.00 ;Setup Jerk
;------------------------------------------------------------------
M500              ; save your preferences to the EEPROM 
M997              ; optional to reboot the printer
```
Information about of the additional custom C g-codes can be found here: https://github.com/mriscoc/Ender3V2S1/wiki/Special-custom-C-gcodes

**Comment by SorenRR** [comment]
Is it possible to create a fake "firmware" that when booted on will dump config on sdcard?

**Reply by mriscoc** [reply]
Yes, is possible to do firmware that only dumps the EEPROM values to create a g-code configuration file and save in the SD card. But you have the M503 to do something similar.
https://marlinfw.org/docs/gcode/M503.html

**Reply by SorenRR** [reply]
So I simply create a gcode file with "M503 C S" in it and print it?

**Reply by SorenRR** [reply]
Hmm... That gcode file with M503 did not seem to work as planned. Putty is always a friend when you need one ;-)
